### PR TITLE
[CHAMP] - refactor reports to knockout and added tests

### PR DESCRIPTION
**Pull request by lwyszomi**
Hi @fyniac 

I refactored the Champ reports to knockout and I added tests for them. Could you assign someone to review?

Please let me know if you have any questions.

Regards,
Łukasz

### CODE REVIEW

**Comment by fyniac** [comment]
@orangejenny can you review for the knockout refactor?

**Comment by orangejenny** [comment]
@fyniac sure

**Comment by orangejenny** [comment]
Test failures look real: they're coming from `custom.champ.tests`

**Comment by lwyszomi** [comment]
@orangejenny All tests pass

File: custom/champ/static/champ/js/knockout/prevision_vs_achievement_graph.js [hunk 1]
```diff
@@ -0,0 +1,341 @@
+/* global ko, $, hqImport, moment, nv, d3 */
+
+ko.bindingHandlers.select2 = {
```
Comment (orangejenny) [inline custom/champ/static/champ/js/knockout/prevision_vs_achievement_graph.js]: Might be worth sharing this code between this file and the other two that use this binding.

Comment (orangejenny) [reply custom/champ/static/champ/js/knockout/prevision_vs_achievement_graph.js]: (not blocking)

**Review by orangejenny — APPROVED** [review]

**Comment by orangejenny** [comment]
Thank you for rewriting all of this! I'm hoping to get us totally off of angular eventually, this brings that goal closer.

File: custom/champ/static/champ/js/knockout/prevision_vs_achievement_graph.js (hunk 1 above)
Comment (lwyszomi) [reply custom/champ/static/champ/js/knockout/prevision_vs_achievement_graph.js]: @orangejenny I moved this to the new file

Comment (orangejenny) [reply custom/champ/static/champ/js/knockout/prevision_vs_achievement_graph.js]: nice, thanks

**Review by orangejenny — APPROVED** [review]

**Pull request merged**